File: hw/basurto/u3/hw23AbstractFactory/hw23AbstractFactory.java/AbstractFactory/src/main/java/ec/edu/espe/AbstractFactory/controller/WidgetController.java

package ec.edu.espe.AbstractFactory.controller;
import ec.edu.espe.AbstractFactory.model.Button;
import ec.edu.espe.AbstractFactory.model.GUIFactory;
import ec.edu.espe.AbstractFactory.model.Menu;

/**
 *
 * @author Esteban Basurto,CodeBreakers,@ESPE
 */
public class WidgetController {
    public String generateWidgets() {
        GUIFactory aFactory = GUIFactory.getFactory();
        
        Button aButton = aFactory.createButton();
        aButton.caption = "Play";
        
        Menu aMenu = aFactory.createMenu();
        aMenu.caption = "File";
        
        return aButton.paint() + "\n" + aMenu.paint();
    }
}

File: hw/tipan/u3/hw24AbstractFactory/AbstractFactoryPython/controller/widgetcontroller.py
from model.gui_factory import GUIFactory

class WidgetController:
    def generate_widgets(self):
        factory = GUIFactory.get_factory()
        
        button = factory.create_button()
        button.set_caption("Play")
        
        menu = factory.create_menu()
        menu.set_caption("File")
        
        result = ""
        result += button.paint()
        result += "\n"
        result += menu.paint()
        
        return result